tests/publiccloud/xfstests.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: XFStests on Public Cloud images
+#
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+use base "publiccloud::basetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use repo_tools 'generate_version';
+use Mojo::UserAgent;
+
+our $root_dir = '/root';
+
+sub run {
+    my ($self)       = @_;
+    my $REG_CODE     = get_required_var('SCC_REGCODE');
+    my $disk_size    = get_var('PUBLIC_CLOUD_DISK_SIZE', '20');
+    my $black_list   = get_var('XFSTESTS_BLACKLIST');
+    my $xfstests_dir = '/tmp/xfstests-dev';
+
+    $self->select_serial_terminal;
+
+    assert_script_run('curl ' . data_url('publiccloud/xfstests/partitions.sh') . ' -o /tmp/partitions.sh');
+    assert_script_run('curl ' . data_url('publiccloud/xfstests/run.sh') . ' -o /tmp/run.sh');
+
+    my $provider = $self->provider_factory();
+    my $instance = $self->{my_instance} = $provider->create_instance(use_extra_disk => {size => $disk_size});
+    $instance->wait_for_guestregister();
+
+    $instance->scp('/tmp/partitions.sh', 'remote:' . '/tmp/partitions.sh');
+    $instance->scp('/tmp/run.sh',        'remote:' . '/tmp/run.sh');
+    $instance->run_ssh_command(cmd => "sudo chmod +x  /tmp/partitions.sh");
+    $instance->run_ssh_command(cmd => "sudo chmod +x  /tmp/run.sh");
+
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $REG_CODE,         timeout => 600) if (get_required_var('FLAVOR') !~ m/On-Demand/);
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -p sle-sdk/12.5/x86_64 ', timeout => 300);
```

Comment:
Hard code 12.5 and x86_64 better to change to use FAVOR(adjust needed) and ARCH.

Reply:
sure, I was testing on 12.5 and forgot to make it generic.